### PR TITLE
Fixed some bugs

### PR DESCRIPTION
**Pull request by 1123852253**
FIX: GoTo Declaration error （When there are multiple identical keywords in the same file）
fix: Goto Forward invalid
ADD: uchardet encoding detector library (WIN64)

### CODE REVIEW

File: LiteEditor/CodeLiteIDE.project [hunk 1]
```diff
@@ -2109,7 +2164,7 @@ resources.cpp: resources.xrc
         <SearchPaths>../sdk/clang/include</SearchPaths>
       </Completion>
     </Configuration>
-    <Configuration Name="Win_x64_Release" CompilerType="g++-64" DebuggerType="GNU gdb debugger" Type="Executable" BuildCmpWithGlobalSettings="append" BuildLnkWithGlobalSettings="append" BuildResWithGlobalSettings="append">
```
Comment (eranif) [inline LiteEditor/CodeLiteIDE.project]: Can you please rename your local compiler from `MinGW ( mingw64 )` to `g++-64` to 
avoid this change?

**Comment by 1123852253** [comment]
needed
debian/ubuntu： sudo apt install libuchardet-dev

Used here：
CMakeLists.txt: 286
Cl_editor.cpp:wxFontEncoding clEditor::DetectEncoding(const wxString& filename)
Use it to detect text encoding types

**Comment by eranif** [comment]
What about other distros (e.g. CentOS) and Mac?
Will this break the build on these machines?

If you are not sure how to handle this on other platforms, please put your code inside `#ifdef` blocks

**Comment by 1123852253** [comment]
Will not
‘#if defined(USE_UCHARDET)’ is used in the code

**Comment by eranif** [comment]
I don't see any *usage* for this macro in the code in this PR (10 files changed)
if you are making use of this library somewhere else (but not in this PR) please consolidate all the PRs so I can see the big picture. This PR only includes the newly added DLL and a minor fix to the parser code. I can't see the fixes to `CMakeLists.txt`

**Comment by 1123852253** [comment]
code PR https://github.com/eranif/codelite/pull/2098/commits/cbfde9357e084db80f6dfda75c157f4cebf44eb7

**Comment by eranif** [comment]
OK, but it's not part of *this* PR.
Can you please consolidate all the changes into a one big PR?

**Comment by 1123852253** [comment]
Sorry, I don't know how to operate.

**Comment by eranif** [comment]
It's simple: close all the PRs you have submitted and submit ONE PR which includes all your changes

**Comment by 1123852253** [comment]
But the previous PR has been merged  
It has been closed

**Comment by eranif** [comment]
Which PR are you taking about?
The only recent PR I closed from you, is this one: https://github.com/eranif/codelite/pull/2304/files

And it does not contain any changes to `CMakeLists.txt`

**Comment by 1123852253** [comment]
Code and Cmakelist.txt  PR https://github.com/eranif/codelite/pull/2098

**Comment by 1123852253** [comment]
Sorry, I don't know if my changes will be pushed automatically. This is not my intention. It's just a temporary change

I found that some operations have a long response time (>10s) when using python-language-server.
The reason is that the lsp message is too long (1.8M bytes) but ReadAll only reads 1k at a time, I think the message is split too much and affects efficiency.
So I made a modification to verify if it can improve it, it does work but it is not rigorous.

**Comment by eranif** [comment]
How did you test the python language server? can you share a link?
I will most definitely test it locally

**Comment by eranif** [comment]
You might want to change the transport from the default `stdio` to `tcp`.
You can do this by configuring `pyls` to start like this:

From `Plugins -> Language Server -> Settings` :

Change `pyls` to start like this:
`/usr/bin/pyls --host 127.0.0.1 --port 3434 --tcp`

And in the connection string, type:
`tcp://127.0.0.1:3434` - this will tell CodeLite to connect to the python server over TCP

**Comment by 1123852253** [comment]
Pyls test environment
Linuxmint 19.1
Pip3 install setuptools
Pip3 install wheel
Pip3 install python-language-server
Pip3 install opencv-python

Code:
Import cv2
Cv2.#code complete

I tried tcp mode, it can solve this problem.
But only one of them can work when running multiple codelite processes.

**Comment by 1123852253** [comment]
I modified Unixprocess::readall and launched a new push request.

**Pull request closed without merging**